server/src/server.rs

```diff
@@ -349,6 +373,27 @@ pub struct CubebServer {
     shm_area_size: usize,
 }
 
+impl Drop for CubebServer {
+    fn drop(&mut self) {
+        if let Some(device_collection_change_callbacks) = &self.device_collection_change_callbacks {
+            debug!("CubebServer: dropped with device_collection_change_callbacks registered");
+            CONTEXT_KEY.with(|k| {
+                let mut state = k.borrow_mut();
+                if let Some(CubebContextState {
```

Comment:
Is it possible that `state` is `None` here? Doesn't that mean `with_local_context` is never called but we have a `CubebServer`?

Reply:
`with_local_context` is called in `CubebServer::process`, lazily initializing libcubeb on the first call.   Any time initialization fails to reach that point, the `CONTEXT_KEY` `state` will be `None` (e.g. if `bind_server` failed, or the `CubebServer` was disconnected before receiving a message).